recipe/meta.yaml

```diff
@@ -52,6 +54,7 @@ outputs:
     files:
       - include/*.h  # [not win]
       - lib/*.a  # [not win]
+      - lib/libopenblas${SHLIB_EXT}  # [not win]
```

Comment:
I'm still confused @isuruf `libopenblas${SHLIB_EXT}` is a symlink to `libopenblasp-r0.3.6${SHLIB_EXT}`, right? Why they have to be in separate packages? Also, size wise it does not make any difference. I guess that I'm worried that packages searching for `libopenblas${SHLIB_EXT}` will be forced to install the full package instead of just `libopenblas` b/c of a symlink.

Reply:
> libopenblas${SHLIB_EXT} is a symlink to libopenblasp-r0.3.6${SHLIB_EXT}, right? 

Yes

> I guess that I'm worried that packages searching for libopenblas${SHLIB_EXT} will be forced to install the full package instead of just libopenblas b/c of a symlink.

No package will be looking for `libopenblas.so`. They will look for `libopenblas.so.0` which is the SONAME of the library.

Reason for `libopenblas.so` not being in `libopenblas` package is to avoid accidental linking to `libopenblas.so` by other packages. This is what debian and other package managers do. Compare https://packages.debian.org/jessie/amd64/libopenblas-base/filelist and https://packages.debian.org/jessie/amd64/libopenblas-dev/filelist

Reply:
Following that logic should all the symlinks be in `openblas` instead?

Reply:
No. We need the concrete file `libopenblasp-r0.3.6.so` and the SONAME library `libopenblas.so.0` in `libopenblas` because downstream packages will be looking for the SONAME library `libopenblas.so.0` even though it is a symlink.

Reply:
Also, this diverges from what is happening in `defaults`: https://github.com/AnacondaRecipes/openblas-feedstock/blob/master/recipe/meta.yaml#L44

I'm OK with that change but we need to be aware of the package differences, specially b/c we are not building with `strict` and some problems may appear if `defaults` `libopenblas` is selected in our builds! (An important case for us to start using `strict` in our builds ASAP.)

Reply:
Even if `defaults` `libopenblas` is used at runtime, that's fine because they have all the files we have in `libopenblas` and more files. It would be problematic if they had less files.

Reply:
I guess that when you said:

    Reason for libopenblas.so not being in libopenblas package is to avoid accidental
    linking to libopenblas.so by other packages.

I thought that if a package built with `defaults` accidentally links to `libopenblas.so` would it be broken if we try to run with out `libopenblas`?

(I'm not against the change but worried if we should start using `stritc` in the builds or not.)

Reply:
> I thought that if a package built with defaults accidentally links to libopenblas.so would it be broken if we try to run with out libopenblas?

Even if they link accidentally link to `libopenblas.so`, the linked library or executable will look for `libopenblas.so.0` at load time because that is the SONAME of `libopenblas.so`.

Reply:
OK. Then I'm not sure why we need to split it in a different package but since there is no harm let's go with it.

Reply:
> OK. Then I'm not sure why we need to split it in a different package but since there is no harm let's go with it.

To avoid linking to `libopenblas` at all. We want our downstream packages to link to `libblas.so` to allow switching the BLAS implementation

Reply:
> To avoid linking to libopenblas at all. We want our downstream packages to link to libblas.so to allow switching the BLAS implementation

Then I'm back to my other question. Will we be able to switch `blas` if we accidentally use a `libopenblas` from `defaults` at build time?

Reply:
No. You are right

Reply:
I guess that we are still good to merge this as b/c `conda-forge`'s is newer and channel priority will help, but the only way to be really safe is to enable `strict` in our builds ASAP.